src/Interpreters/MonotonicityCheckVisitor.h

```diff
@@ -125,13 +147,55 @@ class MonotonicityCheckMatcher
         }
 
         ColumnsWithTypeAndName args;
-        args.emplace_back(data.arg_data_type, "tmp");
+        ColumnWithTypeAndName const_arg;
+
+        if (arguments_size == 2)
+        {
+            if (ast_function.arguments->children[0]->as<ASTLiteral>())
```

Comment:
This can't be done as is. `if-init` scope is not that smart, it'll conflict with the `else` `literal`. I can use different names or keep it as is, whatever you prefer.